lib/memjs/utils.js

```
@@ -114,10 +114,16 @@ const hashCode = function (str) {
     return Math.abs(ret);
 };
 exports.hashCode = hashCode;
+const ERROR_TOO_MANY_OPEN_CONNECTIONS = "ERROR Too many open connections\r\n";
 const parseMessage = function (dataBuf) {
     if (dataBuf.length < 24) {
         return false;
     }
+    if (dataBuf.length === ERROR_TOO_MANY_OPEN_CONNECTIONS.length) {
```

hmmm, do we need to throw the error and churn stack?  Seems like good way to start, just thinking out loud while sipping coffee

---

good question. I initially wrote it to return ERROR_TOO_MANY_OPEN_CONNECTIONS and then checked for that value in the responseHandler, but it felt a little awkward and like introducing a (super minimal) cost to the happy path for something that should be very rare. The action initiated by the error is rather expensive anyway: fully destroy the socket, reject a bunch of promises, probably cause the error to be thrown by downstream code. I'm interested in your opinion, though.